### ISSUE TITLE
Provide a way to convert `grpc::StatusCode` to string

### ISSUE THREAD

**Issue by rainerschoe** [grpc/grpc#34136]
### Is your feature request related to a problem? Please describe.
In a typical gRPC client implementation one needs to propagate potential RPC failures and/or report them to the user.

While `grpc::Status::error_message()` gives a string description set by the service implementation, the enum value returned by `grpc::Status::error_code()` contains valuable additional information for tracing down a problem.

However, there is currently no way of converting those values of type `grpc::StatusCode` into a human readable form. Which is always frustrating, as I need to write my own lokup table for each client implementation.

### Describe the solution you'd like
A function `std::string to_string(grpc::StatusCode code);` which returns strings like `"CANCELLED"` or `"DEADLINE_EXCEEDED"`.

### Describe alternatives you've considered
We could provide this function as a community third-party library, but I really think this should be part of gRPC C++ API.

### Additional context
This is related to #17969 which was closed with the argument that `error_message()` gives a string. However as described above, I think this is a mis-understanding:
As far as I understand, `error_message()` and `error_code()` transport different kind of information. In the sense that `error_code()` gives a category for the error and `error_message()` gives details. Both are needed for handling a problem.

**Comment by yashykt** [comment]
You could potentially use `absl::StatusCodeToString()`. We are thinking of providing convenience functions to convert grpc::Status to absl::Status

**Comment by rainerschoe** [comment]
> You could potentially use `absl::StatusCodeToString()`. It would need a static cast but it would work.

This would already be helpful. However long-term, I think it is better to hide `absl` from the interface and provide a native `ToString()` function for `grpc::StatusCode` (It could internally just forward to `absl`).

**Comment by yashykt** [comment]
Yeah, I guess ideally, we would want that, but there are a bunch of messy details about status conversions that make it really tricky to add a native API for this at present. Please stay posted, there is work being done around this.. not sure around the timelines though.
